### PR TITLE
opentofu 1.6.0-alpha2 (new formula)

### PR DESCRIPTION
**Pull request by b-reich**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
PR ist WIP, opentofu is still in alpha

### CODE REVIEW

**Comment by Bo98** [comment]
We'll probably wait until the first stable unless there's a desperate need for something earlier, but thanks for getting this going!

File: Formula/o/opentofu.rb
Comment (Bo98) [inline Formula/o/opentofu.rb:18]: What is this about and is it still relevant? Is it shipping a secret `node` somewhere?

Comment (p-linnane) [reply Formula/o/opentofu.rb:18]: Carried over from our terraform formula. Not sure if it's still relevant.

Comment (b-reich) [reply Formula/o/opentofu.rb:18]: Maybe it's because the prebuilt protobuf?

File: audit_exceptions/unstable_allowlist.json
Comment (icco) [inline audit_exceptions/unstable_allowlist.json:8]:
```suggestion
  "opentofu": "1.6.0-alpha2",
```

File: Formula/o/opentofu.rb
Comment (MikeMcQuaid) [reply Formula/o/opentofu.rb:18]: would be good to try and remove it and see if it breaks

Comment (MikeMcQuaid) [reply Formula/o/opentofu.rb:18]: (but can be after this PR is merged)

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
> We'll probably wait until the first stable unless there's a desperate need for something earlier

@Bo98 I reckon there kinda is a desperate need at this point and it's a special-case as we'll 100% be packaging the stable release, despite this not having had one yet.

**Comment by Bo98** [comment]
I don't see how deprecating Terraform is "desperate", and it's not even known if all dependents even support OpenTofu yet given the first alpha release was days ago and it's a separate registry that doesn't have guaranteed uptime yet. But I do agree that I don't have a problem shipping an alpha/beta releases for things that have no stable release.

I would be useful to know if anyone has tested OpenTofu with the dependents we have and if everything works fine.

**Review by p-linnane — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
> I don't see how deprecating Terraform is "desperate",

I don't think it is either but I do think given all the conversation around it right now: it would be more useful to people if this happens at a similar time.

> It would however be useful to know if anyone has tested OpenTofu with the dependents we have and if everything works fine

Agreed.

> an immediate mass migration which would be a stable -> alpha switch.

That's not going to happen until we've verified everything works fine.

**Review by SMillerDev — APPROVED** [review]

File: audit_exceptions/unstable_allowlist.json
Comment (p-linnane) [reply audit_exceptions/unstable_allowlist.json:8]: I've reverted this. We need to leave the last number off since it produces a syntax error and would require us to bump it with every new alpha release.

**Comment by Bo98** [comment]
> That's not going to happen until we've verified everything works fine.

Ok, my proposal then is that any form of migration happens in a single PR rather than several, so we don't have half of Homebrew/core use one and the other half use the other. Given the low number of dependents, this should not be a problem.

**Review by Bo98 — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
> Ok, my proposal then is that any form of migration happens in a single PR rather than several, so we don't have half of Homebrew/core use one and the other half use the other. Given the low number of dependents, this should not be a problem in terms of CI time.

@bo98 Works for me 👍🏻. CC @chenrui333 @iMichka and any other @Homebrew/maintainers who care about this.

**Review by chenrui333 — APPROVED** [review]
🚀 💯

**Pull request merged**
